tests/integration/run_modes/test_simulation.py

```diff
@@ -424,10 +424,12 @@ async def test_settings_broadcast(
 
         # Assert that setting run mode on a simulation mode task fails with
         # warning:
-        schd.broadcast_mgr.put_broadcast(
+        good, bad = schd.broadcast_mgr.put_broadcast(
             ['1066'], ['one'], [{
                 'run mode': 'live',
             }])
+        assert good == []
+        assert bad == {'settings': [{'run mode': 'live'}]}
```

Comment:
If you broadcast an incompatible run mode, the error will now be returned to the client (but will still be logged server-side).

Extended this test to cover the new behaviour.